### ISSUE TITLE
useless-provides debuginfo, rpmlint runs on debuginfo package?

### ISSUE THREAD

**Issue by bkircher** [rpm-software-management/rpmlint#112]
Description of problem:
rpmlint reports
```
botan2-debuginfo.x86_64: E: useless-provides debuginfo(build-id)
```

Version-Release number of selected component (if applicable):
rpmlint version 1.10

How reproducible:
Build following spec:
https://raw.githubusercontent.com/bkircher/botan2-fedora/master/botan2.spec

Steps to Reproduce:
1. download SRPM here: https://github.com/bkircher/botan2-fedora/raw/master/botan2-2.2.0-4.fc26.src.rpm
2. build and run `fedora-review -m fedora-rawhide-x86_64 --rpm-spec -n botan2-2.2.0-4.fc26.src.rpm`
(Takes a while, sorry)

Expected results:
That error is not reported

**Comment by pkovacs** [comment]
I am seeing these rpmlint errors also in the debuginfo rpm's created by Fedora rawhide (f28).   

```
rpm -qpP <package>-debuginfo-<version>.fc28.x86_64.rpm

debuginfo(build-id) = 2cad1344592c40ad45ed0f31641cf17271c0f4e2
debuginfo(build-id) = 3af6a6f4d85c9dea2f7c4ae8d50e3bc54a6e8b7c
debuginfo(build-id) = 3deb0c2a3e119815b7b78e389cd6a3d335f016fd
debuginfo(build-id) = 5357a295e0bb369f6b2b0bef41ae811b09661a13
debuginfo(build-id) = 541a82474c5bcf7c8d5164a19767613607193def
debuginfo(build-id) = 549219a67e1ef179b6a25b19f711fc496ddb04b1
debuginfo(build-id) = 59dd4882196c173e989eeaee4d497ed8ebdaf4d1
debuginfo(build-id) = f6ef0aabe214ca82a3ade337240a240876e618fa
<package>-debuginfo = <version>.fc28
<package>-debuginfo(x86-64) = <version>.fc28
```

rpmlint does not like these debuginfo(build-id) entries and is spitting out the useless-provides error.
You can see the build id in the elf binaries with readelf (below).  rpmlit should not be barking at these.

```
readelf -n <binary>

Displaying notes found in: .note.gnu.build-id
  Owner                 Data size	Description
  GNU                  0x00000014	NT_GNU_BUILD_ID (unique build ID bitstring)
    Build ID: f2da8182d3749defd972813a82d47508f5c9924b
```

**Comment by dirkmueller** [comment]
Isn't that an issue of whatever is invoking rpmlint? if rpmlint is invoked to be run on the -debuginfo package, but you don't actually want that, how about not calling rpmlint on the package then?

the other alternative is to addFilter('^\S+-debuginfo:') which would suppress any warnings from those packages.

**Comment by pkovacs** [comment]
@dirkmueller It might be helpful at least to lower the severity of useless-provides to "warning" rather than "error," so that package maintainers using tools layered on top of rpmlint (e.g. Fedora) can manage their workflow with less friction.  Perhaps the Fedora tooling maintainers could skip over the debuginfo rpms, true, although there may be other errors in those rpms that would be missed as a consequence.    Just a thought.

**Comment by dirkmueller** [comment]
another alternative is that we'll just ignore the warning for stuff that starts with debuginfo(

**Comment by dirkmueller** [comment]
@pkovacs the reason why I think it would be better to skip over debuginfo totally is because there are a million other warnings that don't make sense for debuginfo. for example I have various warnings on unstripped-binary-or-object as well as shared-lib-without-dependency-information on debuginfo.

**Comment by dirkmueller** [comment]
should be fixed in master.
